src/helpers/metrics.ts

```diff
@@ -38,3 +40,49 @@ new client.Gauge({
     this.set((await db.queryAsync(`SELECT SUM(total) as count FROM reqs`))[0].count as any);
   }
 });
+
+// Metrics PushGateway test
+// @todo To be moved to snapshot-metrics package after live testing
+const INSTANCE = process.env.METRICS_INSTANCE;
+const JOB_NAME = process.env.METRICS_JOB_NAME ?? 'prometheus';
+const PUSHGATEWAY_URL = process.env.METRICS_PUSHGATEWAY_URL;
+
+async function sleep(time: number) {
+  return new Promise(resolve => {
+    setTimeout(resolve, time);
+  });
+}
+
+if (PUSHGATEWAY_URL && INSTANCE && JOB_NAME) {
+  const gateway = new client.Pushgateway(PUSHGATEWAY_URL, {
+    timeout: 5e3,
+    agent: new http.Agent({
+      keepAlive: true,
+      keepAliveMsecs: 10e3,
+      maxSockets: 5
+    })
+  });
+
+  function defaultErrorHandler(e: any) {
+    console.error('Error while pushing to Pushgateway', e);
+  }
+
+  async function pushMetrics(errorHandler = defaultErrorHandler) {
+    try {
+      gateway
+        .pushAdd({
+          jobName: JOB_NAME as string,
+          groupings: {
+            instance: `${INSTANCE}:${process.env.HOSTNAME || '80'}`
+          }
+        })
+        .catch(errorHandler);
+    } catch (e) {
+    } finally {
+      await sleep(5e3);
+      await pushMetrics(errorHandler);
+    }
+  }
+
+  pushMetrics(e => capture(e));
```

Comment:
```suggestion
  async function pushMetrics() {
    const metric = {
      jobName: JOB_NAME as string,
      groupings: {
        instance: `${INSTANCE}:${process.env.HOSTNAME || '80'}`
      }
    }
  
    try {
      await gateway.pushAdd(metric)
    } catch (e) {
      capture(e)
    }
  }

  setInterval(pushMetrics, 5e3)
```
better not to use this recursive calling of `pushMetrics` because it potentially could cause a stack overflow

Reply:
Can we keep the errorHandler customisation ? This block will be moved to @snapshot-labs/snapshot-metrics package, which is not aware to sentry